Map controllerv1

JustinWinningham opened the pull request.

JustinWinningham committed:
Map Controller and rendering first pass

Added loading for maps from external files (for user created files)
Map script which loads maps - still need to work on rendering
a few test files and folders - non permanent homes

JustinWinningham committed:
Rendering maps

Able to render a map now :D

JustinWinningham committed:
Refactor Map Loading

Added MAPCUSTODIAN to handle validating and loading maps into memory at launch

Added Map class (Map.gd) for holding map data
Refactored MapBuilder (previously Map.gd) to work with new MAPCUSTODIAN architecture

Added GLOBAL autoload file for global methods outside the scope of MAPCUSTODIAN

Added second testing map

TODO: Validate the data (faking it right now) and cleanup

brooks42 commented on Globals/MAPCUSTODIAN.gd:
```diff
@@ -0,0 +1,92 @@
+
+
+# TODO: Actually validate the data - we are just pretending right now and dealing with the crash
+func validateMapData(filepath):
```
We can do it in a future PR but we should definitely make these enums or something to key off of instead of just chars.

JustinWinningham replied:
Agreed

brooks42 commented on Testing/Map.gd:
```diff
@@ -0,0 +1,34 @@
+# This is a class for maps and all data related to them - namely terrain
```
I don't see anything in the best practices but it did take me a while to find this code in the IDE since this is under `Testing`, not under `Scripts`. If it's not against some practice I vote we consolidate this code into the scripts folder where the other scripts are, like `Scripts/maps/` or something.

JustinWinningham replied:
Moved

brooks42 reviewed (commented):
Just a couple of comments but so far it's looking pretty good.

JustinWinningham committed:
Updated based on PR feedback

Removed testing folder
Relocated files previously located in testing folder to new homes and updated logic that was dependent on testing location

brooks42 reviewed (approved):
LGTM, nice work 👍
